blockservice.go

```diff
@@ -109,23 +92,19 @@ func (s *blockService) Exchange() exchange.Interface {
 // AddBlock adds a particular block to the service, Putting it into the datastore.
 // TODO pass a context into this if the remote.HasBlock is going to remain here.
 func (s *blockService) AddBlock(o blocks.Block) error {
```

Comment:
I think we can remove code duplication here by having AddBlock call AddBlocks.

Reply:
good idea